freqtrade/exchange/__init__.py

```diff
@@ -151,10 +151,10 @@ def buy(pair: str, rate: float, amount: float) -> Dict:
             'Tried to buy amount {} at rate {} (total {}).'
             'Message: {}'.format(pair, amount, rate, rate*amount, e)
         )
-    except ccxt.NetworkError as e:
-        raise NetworkException(
-            'Could not place buy order due to networking error. Message: {}'.format(e)
-        )
+    except (ccxt.NetworkError, ccxt.ExchangeError) as e:
+        raise TemporaryError(
```

Comment:
I think you would want to decorate the function with `@retrier` in that case (or handle the issue downstream) - but personally, i don't think it's a good idea to buy/sell retry - could happen in double-buys / sells if the exchange got the order, but answered "badly"...

Reply:
@xmatthias Good point, I will add the decorator to all exchange functions except `buy` and `sell`. I'm not sure yet what's the best way to deal with those two.